dart/lib/src/event_processor/enricher/html_enricher_event_processor.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:html' as html show window, Window;
+
+import '../../../sentry.dart';
+import 'enricher_event_processor.dart';
+
+EnricherEventProcessor enricherEventProcessor(SentryOptions options) {
+  return WebEnricherEventProcessor(
+    html.window,
+    options,
+  );
+}
+
+class WebEnricherEventProcessor implements EnricherEventProcessor {
```

Comment:
Should be called `HtmlEnricherEventProcessor`?

In any case, the difference between `html` and `web` implementations is only the access to `http.Window/web.Window` properties, the rest is all duplicated code.

Would it make sense to move these properties into a common interface/class and use it based in conditional import?

Reply:
> Should be called `HtmlEnricherEventProcessor`?

The name is asserted in tests so I kept it the same
```dart

    test('$WebEnricherEventProcessor gets added on init', () async {
      late SentryOptions sentryOptions;
      await Sentry.init(
        (options) {
          options.dsn = fakeDsn;
          sentryOptions = options;
        },
      );
      await Sentry.close();

      expect(sentryOptions.eventProcessors.map((e) => e.runtimeType.toString()),
          contains('$WebEnricherEventProcessor'));
    });
```


> In any case, the difference between `html` and `web` implementations is only the access to `http.Window/web.Window` properties, the rest is all duplicated code.
> 
> Would it make sense to move these properties into a common interface/class and use it based in conditional import?

I started doing that last week but then noticed there are more changes in actual implementations so I don't think it makes much sense to extract the common functions. See diff:
```diff
1c1,3
< import 'dart:html' as html show window, Window;
---
> // We would lose compatibility with old dart versions by adding web to pubspec.
> // ignore: depend_on_referenced_packages
> import 'package:web/web.dart' as web show window, Window, Navigator;
8c10
<     html.window,
---
>     web.window,
19c21
<   final html.Window _window;
---
>   final web.Window _window;
62,65c64,66
<       screenHeightPixels: device?.screenHeightPixels ??
<           _window.screen?.available.height.toInt(),
<       screenWidthPixels:
<           device?.screenWidthPixels ?? _window.screen?.available.width.toInt(),
---
>       screenHeightPixels:
>           device?.screenHeightPixels ?? _window.screen.availHeight,
>       screenWidthPixels: device?.screenWidthPixels ?? _window.screen.availWidth,
80,87c81,83
<     final screenOrientation = _window.screen?.orientation;
<     if (screenOrientation != null) {
<       if (screenOrientation.type?.startsWith('portrait') ?? false) {
<         return SentryOrientation.portrait;
<       }
<       if (screenOrientation.type?.startsWith('landscape') ?? false) {
<         return SentryOrientation.landscape;
<       }
---
>     final screenOrientation = _window.screen.orientation;
>     if (screenOrientation.type.startsWith('portrait')) {
>       return SentryOrientation.portrait;
88a85,87
>     if (screenOrientation.type.startsWith('landscape')) {
>       return SentryOrientation.landscape;
>     }
103a103,106
> 
> extension on web.Navigator {
>   external double? get deviceMemory;
> }
```

Reply:
Makes sense, no reason then to do this. 👍